crates/matrix-sdk-crypto/src/olm/session.rs

```diff
@@ -146,26 +150,59 @@ impl Session {
         content: impl Serialize,
         message_id: Option<String>,
     ) -> OlmResult<Raw<ToDeviceEncryptedEventContent>> {
+        #[derive(Debug)]
+        struct Content<'a> {
+            event_type: &'a str,
+            content: Raw<Value>,
+        }
+
+        impl EventType for Content<'_> {
+            // This is a bit of a hack, usually we just define the `EVENT_TYPE` and use the
+            // default implementation of `event_type()`, we can't do this here
+            // because the event type isn't static.
```

Comment:
multi-way comma splice!

```suggestion
            // This is a bit of a hack: usually we just define the `EVENT_TYPE` and use the
            // default implementation of `event_type()`. We can't do this here
            // because the event type isn't static.
```